pkg/storage/etcd/etcd_helper.go

```diff
@@ -72,18 +72,33 @@ type EtcdConfig struct {
 	CertFile   string
 	CAFile     string
 	Quorum     bool
+	Username   string
+	Password   string
 }
 
-func (c *EtcdConfig) newEtcdClient() (etcd.Client, error) {
+func (c *EtcdConfig) newClientConfig() (*etcd.Config, error) {
```

Comment:
This was because `c.newHttpTransport()` could return `nil` on error here. But we could construct the transport in `newEtcdClient()` and just pass as an arg to `newClientConfig()`?